accounts/service.go

```diff
@@ -529,11 +561,32 @@ func (s *InterceptorService) TrackPayment(id AccountID, hash lntypes.Hash,
 		return nil
 	}
 
-	// Okay, we haven't tracked this payment before. So let's now associate
-	// the account with it.
-	account.Payments[hash] = &PaymentEntry{
-		Status:     lnrpc.Payment_UNKNOWN,
-		FullAmount: fullAmt,
+	if !ok {
+		// In the rare case that the payment isn't associated with an
+		// account yet, we do that first, to ensure that we will track
+		// the payment in case the service is restarted. This can happen
+		// if the payment is a KeySend or SendToRoute payment.
+		// This is attempted even if the service is disabled, as we'll
+		// then track the payment on next startup.
+		account.Payments[hash] = &PaymentEntry{
```

Comment:
Why are we putting this behind the `!ok` flag now? I think we always want to reset the status to `lnrpc.Payment_UNKNOWN` if we call `TrackPayment`, to avoid missing any updates.

Reply:
Ah sorry! Didn't understand that was the intention, I've fixed this now with the latest push!